doc/compiler/optimizer/ValuePropagation.md

```diff
@@ -0,0 +1,195 @@
+
+### Absolute vs. Relative Constraints
+
+Absolute constraints are explicit constraints about a particular value number. For example a type constraint, a range constraint, a null constraint are all examples or absolute constraints.
+
+Relative constraints are constraints that connect one value number to another value number. For example if we were to encounter a guarded comparison between `x` and `y`, say `x < y` then we would create a relative constraint between the value numbers of `x` and `y` to specify that on that path the value number for `x` is less than the value number for `y`.
+
+Relative constraints are useful when neither `x` nor `y` are known to be constant, and we still want to propagate information about the relation of their value numbers. This is useful if for example we encounter another condition which is a successor of the previous check which asked if `x < y + 5`. We want to be able to fold away such expressions even though we don't know any absolute constraints about the value numbers of `x` nor `y`.
+
+## Local vs. Global Value Propagation
+
+There are two modes in which the value propagation optimization can run; local and global. Both modes run using the same constraints and VP handlers. What is different is the scope at which they operate.
+
+Local value propagation processes the program one block at a time. If block extension has ran it processes the program one extended block at a time. Information is not propagated between blocks, so any information we learned is discarded after arriving at the end of the (extended) basic block. This of course makes the local value propagation optimization restrictive in what it can accomplish.
+
+Because local value propagation runs at this reduced scope there is no need to calculate value numbers for all the nodes in the program. We can simply use the global index of the node as the value number, since if local value propagation runs after local common subexpression elimination (CSE), the nodes within a block will have already been commened up and they will have the same global index. Retrieving the global index is very cheap, and in addition other expensive metadata such as use-def info need not be computed to run local value propagation.
+
+This makes it the perfect candidate to run frequently and at low optimization levels.
+
+Global value propagation on the other hand is a global optimization which requires both global value numbering and use-def information to be computed. In the context of the OMR compiler global means at the scope of the current compilation unit, not intra-procedure. The extra metadata information which needs to be computed is relatively expensive and this is why the global value propagation optimization is only run at higher optimization levels (warm and above). 
+
```

Comment:
I think you mean "inter procedural" when you said "intra-procedure"

Reply:
Fixed in [force-push](https://github.com/eclipse/omr/compare/5c1aa757ca3e116307e60404b9d3b0743e2ce03f..127bdeabc6860a9f211446b05d6210a084512605).